process/smartContract/hooks/factory.go

```diff
@@ -10,14 +10,22 @@ import (
 
 // CreateBlockChainHook creates a blockchain hook based on the chain run type (normal/sovereign)
 func CreateBlockChainHook(chainRunType common.ChainRunType, args ArgBlockChainHook) (process.BlockChainHookHandler, error) {
-	bh, err := NewBlockChainHookImpl(args)
-
+	factory, err := NewBlockChainHookFactory()
+	if err != nil {
+		return nil, err
+	}
 	switch chainRunType {
 	case common.ChainRunTypeRegular:
-		return bh, err
+		args.BlockChainHookFactoryHandler = factory
 	case common.ChainRunTypeSovereign:
-		return NewSovereignBlockChainHook(bh)
+		sovereignFactory, sovErr := NewSovereignBlockChainHookFactory(factory)
+		if sovErr != nil {
+			return nil, sovErr
+		}
+		args.BlockChainHookFactoryHandler = sovereignFactory
```

Comment:
function that alters the arguments. Please refactor. Why do we still need this function `CreateBlockChainHook`? We should use directly the instances of type  BlockChainHookFactoryHandler

Reply:
the entire switch will be removed in a future PR. To keep everything working, just the factories were created and used.